protocols/horizon/main.go

```
@@ -47,6 +47,28 @@ type Account struct {
 	Data                 map[string]string `json:"data"`
 }
 
+// GetNativeBalance returns the native balance of the account
+func (a Account) GetNativeBalance() string {
+	for _, balance := range a.Balances {
+		if balance.Asset.Type == "native" {
+			return balance.Balance
+		}
+	}
+
+	return "0"
```

since this is native I think we should throw an error instead of returning 0. (whereas 0 is fine for credit balance below)

---

yeah this might deserve a panic()

---

Panicking isn't the right solution.  Instances of this struct are often going to be created and populated from remote data and we shouldn't be crashing people's programs because we don't properly protect against the diversity of remote responses.  The fact of the matter is that the validity of any `Account` value is not only dependent upon it being "Valid as JSON", but also on whether or not it is "Valid as a Stellar Account".  The latter idea requires for example that an account have a native balance (you might even go a little deeper and say an account must hold a valid minimum reserve), but unfortunately you cannot guarantee that requirement is fulfilled just given a `[]Balance`. Invalid values will occur. So, what is our alternative to crashing?  We provide some mechanism to signal the invalidity to our downstream users.  

The question then becomes what is the appropriate mechanism to signal this invalid state.  I see two main potentials:  Report an error at time of access, or report an error at time of instantiation.  The former is simple:  `GetNativeBalance() string` becomes `GetNativeBalance() (string, error)`.  The latter possibility is trickier, but could offer an useful API that we might pursue.  I could see such a system looking similar to how we validate application configs: First we parse and load the struct, then we send it through a secondary validation system to ensure the value is valid for use.  Seems complicated, but it would allow for us to define custom and rich validation schemes for values. 

---

Of note:  The "array of objects" design for balances was chosen so that horizon's protocol was future-compatible with any new asset types that might occur.  Additionally, I liked the symmetry of the design in which the native balance wasn't privileged over the other assets, but that is less important to me.  It's possible that we'll have an opportunity soon to introduce a "v2" of the horizon API and this is one area where we might want to improve things.  I'm open to proposals that keep the notion of being forward compatible with new and exotic asset types but provide a simpler API.

---

Thanks for the detailed options and the rationale explanation, it's super helpful.

I changed the signature to be `GetNativeBalance() (string, error)` to preserve backward compatibility for now.

---

that should be fine. when someone upgrades to the new version of the sdk their compiler will require them to also handle this change.